dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/processor/queue/StateEventResponseService.java

```diff
@@ -142,18 +146,14 @@ private void persist(StateEvent stateEvent) {
                     break;
                 default:
             }
-            workflowExecuteThread.addStateEvent(stateEvent);
+            workflowExecuteThreadPool.submitStateEvent(stateEvent);
```

Comment:
events in one workflow need to be handled in one thread.

Reply:
I keep the event loop logic for it. Please review again.